remsfal-service/src/main/java/de/remsfal/service/control/PropertyController.java

```diff
@@ -83,4 +113,123 @@ public boolean deleteProperty(final String projectId, final String propertyId) {
         return propertyRepository.deletePropertyById(projectId, propertyId) > 0;
     }
 
+    public List<ProjectTreeNodeModel> getProjectTree(
+            final String projectId, final Integer offset, final Integer limit) {
+        logger.infov("Retrieving up to {1} properties (projectId = {0})", projectId, limit);
+
+        // Fetch properties for the project
+        List<PropertyEntity> properties = propertyRepository.findPropertiesByProjectId(projectId, offset, limit);
+
+        return properties.stream()
+                .map(this::buildPropertyNode)
+                .collect(Collectors.toList());
+    }
+
+    private ProjectTreeNode buildPropertyNode(PropertyEntity property) {
+        List<BuildingEntity> buildings = buildingRepository.findBuildingByPropertyId(property.getId());
+        List<SiteEntity> sites = siteRepository.findSiteByPropertyId(property.getId());
+
+        NodeData propertyData = new NodeData(
+                "property",
+                property.getTitle(),
+                property.getDescription(),
+                "",
+                0
+        );
+
+        ProjectTreeNode propertyNode = new ProjectTreeNode(property.getId(), propertyData);
+
+        float sumUsableSpace = 0;
+
+        // Add building nodes
+        for (BuildingEntity building : buildings) {
+            ProjectTreeNode buildingNode = buildBuildingNode(building);
+            propertyNode.addChild(buildingNode);
+            sumUsableSpace += building.getUsableSpace();
+        }
+
+        // Add site nodes
+        for (SiteEntity site : sites) {
+            System.out.println(site.getId());
```

Comment:
This might be replaced by a logger if any.

Reply:
I forgot to remove this line. Thanks for pointing it out.